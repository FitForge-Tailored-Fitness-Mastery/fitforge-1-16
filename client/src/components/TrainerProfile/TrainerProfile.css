.trainer-profile-container {
    max-width: 400px;
    margin: 20px auto;
    padding: 20px;
    background-color: #333;
    color: white;
    font-family: Arial, sans-serif;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
  
  .trainer-profile-header {
    text-align: center;
    margin-bottom: 20px;
  }
  
  .trainer-profile-picture-container {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    overflow: hidden;
    margin-bottom: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  
  .trainer-profile-photo {
    max-width: 100%;
    max-height: 100%;
    object-fit: cover;
  }
  
  .trainer-profile-name {
    color: gold;
  }
  
  .trainer-profile-details {
    width: 100%;
    text-align: left;
    font-size: 0.9rem;
    padding: 10px;
    border-top: 1px solid #555;
  }
  
  .trainer-profile-details a {
    color: gold;
    text-decoration: none;
  }
  
  .trainer-profile-details a:hover {
    text-decoration: underline;
  }
  
  .error {
    color: red;
    font-weight: bold;
    margin-bottom: 10px;
  }
  