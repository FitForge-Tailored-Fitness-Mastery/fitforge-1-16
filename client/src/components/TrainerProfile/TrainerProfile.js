import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import './TrainerProfile.css';

const TrainerProfile = () => {
  const [trainerDetails, setTrainerDetails] = useState(null);
  const [error, setError] = useState(null);
  const { clientId } = useParams(); // Destructure clientId from params

  useEffect(() => {
    // Function to fetch trainer details
    const fetchTrainerDetails = async () => {
      try {
        const response = await fetch(`http://localhost:5000/trainer/${clientId}`);
        if (!response.ok) {
          throw new Error('Network response was not ok');
        }
        const data = await response.json();
        setTrainerDetails(data);
      } catch (error) {
        console.error('Error:', error);
        setError(error.message);
      }
    };

    fetchTrainerDetails();
  }, [clientId]); // Dependency array includes clientId to refetch when it changes

  if (!trainerDetails && !error) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  return (
    <div className="trainer-profile-container">
      <div className="trainer-profile-header">
        {/* Image and name of the trainer */}
        <img src={trainerDetails?.imageUrl} alt={trainerDetails?.fname} className="trainer-profile-photo" />
        <h2 className="trainer-profile-name">{trainerDetails?.fname} {trainerDetails?.lname}</h2>
      </div>
      <div className="trainer-profile-details">
        <p>Email: {trainerDetails?.email}</p>
        <p>Phone: {trainerDetails?.phone}</p>
        {/* Other trainer details */}
      </div>
    </div>
  );
};

export default TrainerProfile;
