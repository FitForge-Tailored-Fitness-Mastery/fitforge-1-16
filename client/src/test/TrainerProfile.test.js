import renderer from 'react-test-renderer';
import React from 'react';
import TrainerProfile from '../components/TrainerProfile/TrainerProfile';

it('testing the testing feature', () => {
    const tree = renderer.create(<TrainerProfile />).toJSON();
    expect(tree).toMatchSnapshot();
  });