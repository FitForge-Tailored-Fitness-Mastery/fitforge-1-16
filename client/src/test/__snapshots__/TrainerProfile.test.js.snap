// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the testing feature 1`] = `
<div>
  Loading...
</div>
`;
